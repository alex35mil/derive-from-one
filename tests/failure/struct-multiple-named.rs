use derive_from_one::FromOne;

#[derive(FromOne)]
pub struct T {
    a: u32,
    b: bool,
}

fn main() {
    let _: T = T::from(1);
}
