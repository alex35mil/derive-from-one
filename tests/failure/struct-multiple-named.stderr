error: FromOne can only be used with structs with a single field
 --> tests/failure/struct-multiple-named.rs:4:12
  |
4 | pub struct T {
  |            ^

error[E0308]: mismatched types
  --> tests/failure/struct-multiple-named.rs:10:24
   |
10 |     let _: T = T::from(1);
   |                ------- ^ expected `T`, found integer
   |                |
   |                arguments to this function are incorrect
   |
note: associated function defined here
  --> $RUST/core/src/convert/mod.rs
   |
   |     fn from(value: T) -> Self;
   |        ^^^^
